Creating-applications-with-smart-card-support.md

```diff
@@ -1,92 +1,98 @@
-* "Mac Dev Center: Security Overview: Security Architecture":http://developer.apple.com/mac/library/documentation/Security/Conceptual/Security_Overview/Architecture/Architecture.html
 
-h3. Java Cryptography Architecture (JCA)
+* [Mac Documentation: Security](https://developer.apple.com/documentation/security/)
 
-* [[Using-smart-cards-with-Java-SE]]
-* Sample "applet":http://emergya.github.com/opensc-testing/ with a "JCA provider":http://docs.oracle.com/javase/1.5.0/docs/api/java/security/Provider.html using OpenSC. ("source":http://github.com/emergya/opensc-testing "javadoc":http://emergya.github.com/opensc-testing/apidocs/index.html)
+### Java Cryptography Architecture (JCA)
 
-h2. SSL/TLS toolkit integration
+* [Using smart cards with Java SE](Using-smart-cards-with-Java-SE)
+* Sample [applet](http://emergya.github.com/opensc-testing/) with a [JCA provider](http://docs.oracle.com/javase/1.5.0/docs/api/java/security/Provider.html) using OpenSC.
+  * [source](http://github.com/emergya/opensc-testing)
+  * [javadoc](http://emergya.github.com/opensc-testing/apidocs/index.html)
 
+## SSL/TLS toolkit integration
 
-h3. OpenSSL
+### OpenSSL
 
-"OpenSSL":http://www.openssl.org/ has an easy way to integrate smart card support. "libp11":http://www.opensc-project.org/libp11/ has code to make using OpenSC PKCS#11 module with OpenSSL quite easy and includes example code for using SSL with client certificate authentication using a smart card too (FIXME: link). "engine_pkcs11":http://www.opensc-project.org/engine_pkcs11 project has an OpenSSL engine implementation so you can change any code using OpenSSL to move the crypto operation from your CPU to your smart card with only a few small changes. Wpa_supplicant is an example of an application using OpenSSL and this engine for smart card support.
+[OpenSSL](http://www.openssl.org/) has an easy way to integrate smart card support. The [libp11](https://github.com/OpenSC/libp11/wiki) has code to make using OpenSC PKCS#11 module with OpenSSL quite easy and includes example code for using SSL with client certificate authentication using a smart card too. The [engine_pkcs11](http://www.opensc-project.org/engine_pkcs11) project has an OpenSSL engine implementation so you can change any code using OpenSSL to move the crypto operation from your CPU to your smart card with only a few small changes. Wpa_supplicant is an example of an application using OpenSSL and this engine for smart card support.
```

Comment:
Maybe also this one extending a bit to mention that OpenSSL <3.0 uses the engines (which is not provided by libp11) and for newer openssl we recommend using pkcs11 provider.

The link here `http://www.opensc-project.org/engine_pkcs11` is dead too.